### PR TITLE
Use external speedyj repo

### PR DESCRIPTION
**Pull request by johannes-wolf**
### Changes

* Remove `speedyj` from the repo.
* Use cmakes `FetchContent` for pulling `Catch2`, `stx` and `speedyj`.
* CI build with zserio 2.3.0
* Migrate tests from gtest to Catch2
* Replace nullptr-casts with `std::declval<>`

### Review Checklist

- [x] The changes are understood.
- [x] The changes are well documented.
- [x] The changes have been tested.

### CODE REVIEW

File: runtime/test/zserio/json_test/json_test.cpp [hunk 1]
```diff
@@ -4,67 +4,66 @@
...
-
-#include <zsr/speedy-j.hpp>
+#include "speedyj/speedyj.hpp"
 
 namespace {
 
 PKG;
 
-TEST(JsonTest, empty_object)
+TEST_CASE("empty_object", "[JsonTest::empty_object]")
```
Comment (josephbirkner) [inline runtime/test/zserio/json_test/json_test.cpp]: Test should move to speedyj...

File: runtime/test/zserio/json_test/json_test.cpp [hunk 2]
```diff
@@ -4,67 +4,66 @@
...
 {
     speedyj::Stream ss;
     ss << speedyj::Object << speedyj::End;
 
-    ASSERT_EQ(ss.str(), R"({})");
+    REQUIRE(ss.str() == R"({})");
 }
 
-TEST(JsonTest, empty_array)
+TEST_CASE("empty_array", "[JsonTest::empty_array]")
```
Comment (josephbirkner) [inline runtime/test/zserio/json_test/json_test.cpp]: Test should move to speedyj...

File: runtime/test/zserio/json_test/json_test.cpp [hunk 3]
```diff
@@ -4,67 +4,66 @@
...
 {
     speedyj::Stream ss;
     ss << speedyj::Array << speedyj::End;
 
-    ASSERT_EQ(ss.str(), R"([])");
+    REQUIRE(ss.str() == R"([])");
 }
 
-TEST(JsonTest, key_value_pair)
+TEST_CASE("key_value_pair", "[JsonTest::key_value_pair]")
```
Comment (josephbirkner) [inline runtime/test/zserio/json_test/json_test.cpp]: Test should move to speedyj...

File: runtime/test/zserio/json_test/json_test.cpp [hunk 4]
```diff
@@ -4,67 +4,66 @@
...
     ss << speedyj::Object
        << "key" << "value"
        << speedyj::End;
 
-    ASSERT_EQ(ss.str(), R"({"key":"value"})");
+    REQUIRE(ss.str() == R"({"key":"value"})");
 }
 
-TEST(JsonTest, json_string_escaping)
+TEST_CASE("json_string_escaping", "[JsonTest::json_string_escaping]")
```
Comment (josephbirkner) [inline runtime/test/zserio/json_test/json_test.cpp]: Test should move to speedyj...

File: runtime/test/zserio/json_test/json_test.cpp [hunk 5]
```diff
@@ -4,67 +4,66 @@
...
     ss << speedyj::Object
        << "\"hello\"" << "\\world\\"
        << speedyj::End;
 
-    ASSERT_EQ(ss.str(), "{\"\\\"hello\\\"\":\"\\\\world\\\\\"}");
+    REQUIRE(ss.str() == "{\"\\\"hello\\\"\":\"\\\\world\\\\\"}");
 }
 
-TEST(JsonTest, json_uchar_string_escaping)
+TEST_CASE("json_uchar_string_escaping", "[JsonTest::json_uchar_string_escaping]")
```
Comment (josephbirkner) [inline runtime/test/zserio/json_test/json_test.cpp]: Test should move to speedyj...

**Review by josephbirkner — APPROVED** [review]
Pretty damn nice. Some tests should be moved to speedyj I think.

**Comment by johannes-wolf** [comment]
> Pretty damn nice. Some tests should be moved to speedyj I think.

Going to remove them, as they are already in speedyj.

**Pull request merged**